src/hooks/useCart/useCart.tsx

```diff
@@ -128,6 +140,44 @@ const mergeWithCart = (
   );
 };
 
+export const findOptionalIdentifierInputLabels = (
+  policies: CampaignPolicy[]
+): string[] => {
+  const modifiedPolicyIdentifiers: Array<
+    Array<ModifiedPolicyIdentifier>
+  > = policies
+    .map(({ category, identifiers }: CampaignPolicy) => ({
+      category,
+      identifiers,
+    }))
+    .filter(({ identifiers }) => !!identifiers)
+    .filter((groupedIdentifiers): groupedIdentifiers is {
+      category: string;
+      identifiers: PolicyIdentifier[];
+    } => {
+      const { identifiers } = groupedIdentifiers;
+      return !!identifiers;
+    })
+    .map(
```

Comment:
Would it be better if we filtered identifiers that are optional before we craft the final label?